cordapp/src/main/kotlin/com/r3/corda/finance/obligation/client/flows/SendToSettlementOracle.kt

```diff
@@ -42,9 +43,13 @@ class SendToSettlementOracle(
         val obligationState = obligationStateAndRef.state.data
         val settlementMethod = obligationState.settlementMethod as OffLedgerPayment<*>
 
+        requireThat {
+            "Settlement Oracle is not null" using (settlementMethod.settlementOracle != null)
```

Comment:
this could be a little clearer perhaps. "Settlement oracle must not be null"

Reply:
if you use an if statement here, then you can get a smart cast of settlementOracle from `Party?` to `Party`

Reply:
OK